### PR TITLE
[SPARK-16778][SQL][TRIVIAL] Fix deprecation warning with SQLContext

### PR DESCRIPTION
**Pull request by holdenk**
## What changes were proposed in this pull request?

Change to non-deprecated constructor for SQLContext.
## How was this patch tested?

Existing tests

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala [hunk 1]
```diff
@@ -28,7 +28,7 @@ class SQLConfSuite extends QueryTest with SharedSQLContext {
   test("propagate from spark conf") {
     // We create a new context here to avoid order dependence with other tests that might call
     // clear().
-    val newContext = new SQLContext(sparkContext)
+    val newContext = new SQLContext(SparkSession.builder().sparkContext(sc).getOrCreate())
```
Comment (srowen) [inline sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala]: sc has to be `sparkContext` here, no? we're about to find out from Jenkins

**Comment by srowen** [comment]
Jenkins test this please

File: sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala (hunk 1 above)
Comment (holdenk) [inline sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala]: oops thought this built locally but I guess I was on the wrong branch.

**Comment by srowen** [comment]
OK that looks reasonable

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**